src/Radicle/Daemon/Client.hs

```diff
@@ -0,0 +1,99 @@
+-- | Client for the daemons machine API.
+--
+-- Also provides primitive functions to bind the client into the Radicle
+-- interpreter with @daemonClientPrimFns@.
+module Radicle.Daemon.Client
+    ( daemonClientPrimFns
+    , newMachine
+    , query
+    , send
+    , MachineId(..)
+    ) where
+
+import           Protolude hiding (TypeError)
+
+import           Control.Monad.Except
+import           GHC.Exts (fromList)
+import qualified Network.HTTP.Client as HttpClient
+import           Servant.Client hiding (Client)
+
+import           Radicle
+import           Radicle.Daemon.HttpApi
+import qualified Radicle.Internal.PrimFns as PrimFns
+
+
+-- | Constraints for all client functions
+type Client m = (MonadIO m, MonadError ServantError m)
+
+-- | Send an input to a machine to be evaluated and applied. Returns the results
+-- of evaluting the inputs.
+send :: Client m => MachineId -> Seq Value -> m [Value]
+send machineId inputs = do
+    SendResponse {..} <- runClient $ client machineSendEndpoint machineId (SendRequest $ toList inputs)
```

Comment:
Can `runClient`, `result`, and `wrapServantError` be combined into a function `foo` so we get:
```
query :: (Client m) => MachineId -> Value -> Lang m Value
query machineId q = foo $ client machineQueryEndpoint machineId (QueryRequest q)
```
and likewise for `send` and `newMachine`?

Reply:
`exe/Machines` does not care about the `Lang` monad so an interface with less assumptions prevents unnecessary unwrapping. In general I think using `Client m` rather than `Lang m` makes the interface more flexible.